pkg/volume/snapshot_params.go

```diff
@@ -0,0 +1,98 @@
+package volume
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/piraeusdatastore/linstor-csi/pkg/linstor"
+)
+
+//go:generate enumer -type=SnapshotType -trimprefix=SnapshotType
+
+type SnapshotType int
+
+const (
+	SnapshotTypeInCluster SnapshotType = iota
+	SnapshotTypeS3
+	SnapshotTypeLinstor
+)
+
+type SnapshotParameter struct {
```

Comment:
Singular "Parameter" for the storing the plural "parameters" of the VolumeSnapshotClass?